packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -129,7 +129,9 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
         assert(deleted, "deleted range must exist in forest");
     }
     allocateCursor(): Cursor {
-        return new Cursor(this);
+        const cursor = new Cursor(this);
+        cursor.set(this.rootField, 0);
```

Comment:
One issue cursor has is it has to point to a node, this means that if you have a currently empty, this will fail. I'm not sure if it throws or silently leaves the cursor in a cleared state, but it's not good.

Instead, the caller of allocateCursor should call forest.tryGet(forest.root(forest.rootField), theCursor) if they want to read the root and check the returned value. This allows detecting and handling that there is currently no root node. An example of this is in the object-forest tests.